units/generate_tasks.go

```diff
@@ -0,0 +1,116 @@
+package units
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/evergreen-ci/evergreen/model"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/evergreen-ci/evergreen/validator"
+	"github.com/evergreen-ci/gimlet"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+	mgo "gopkg.in/mgo.v2"
+)
+
+const (
+	generateTasksJobName = "event-send"
+)
+
+func init() {
+	registry.AddJobType(generateTasksJobName, func() amboy.Job { return makeGenerateTaskJob() })
+}
+
+type generateTasksJob struct {
+	job.Base `bson:"job_base" json:"job_base" yaml:"job_base"`
+	TaskID   string            `bson:"task_id" json:"task_id" yaml:"task_id"`
+	JSON     []json.RawMessage `bson:"json" json:"json" yaml:"json"`
+}
+
+func makeGenerateTaskJob() *generateTasksJob {
+	j := &generateTasksJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    generateTasksJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+
+	return j
+}
+
+func NewGenerateTaskJob(id string, json []json.RawMessage) amboy.Job {
+	j := makeGenerateTaskJob()
+	j.TaskID = id
+	j.JSON = json
+
+	j.SetID(fmt.Sprintf("%s:%s", generateTasksJobName, id))
+	return j
+}
+
+func (j *generateTasksJob) Run(ctx context.Context) {
+	defer j.MarkComplete()
+
+	projects, err := ParseProjects(j.JSON)
```

Comment:
why is parse project public and in this package?

Reply:
Made private. It's convenient to keep here because its test data overlap substantially with the other test data in this package.